refactor(list): add docs and improve list cmd (#8)
dhis2/cli-utils-codemods: docs/guides/adding_a_codemod.md

# Adding a codemod

If you need to add a discoverable codemod, then you need to

> 1. figure out the correct repository<br />

Reviewer: ```suggestion
1. determine the correct repository<br />
```

   If you want to provide a codemod for the `app-runtime` library, then the
   library's repository is the correct location

> 1. put the file into a `codemods` folder on the root level<br />

Reviewer: ```suggestion
2. put the file into a `codemods` folder at the root level of the package<br />
```

Author: I changed the test, I kept the `.1` as markdown automatically increments it, so it's less hassle to add items somewhere inbetween

   If the repository is a monorepo, make sure to put the folder into the root

>    folder of the specific workspace as it wouldn't be published otherwise

Reviewer: ```suggestion
   folder of the specific workspace otherwise it will not be published
```

> 

Reviewer: ```suggestion
3. ensure that the `codemods` folder is published to `npm` with the package<br/>
    The `files` list in `package.json` must include `codemods`, and it should not
    be excluded by `.npmignore`

```

> Currently we only support library's that are within the `@dhis2` namespace,

Reviewer: ```suggestion
Currently we only support libraries within the `@dhis2` namespace on `npm`,
```

> such as `@dhis2/cli-utils-cypress` or `@dhis2/ui-core`

Author: Oh, yeah. I could make it a command option that defaults to '@dhis2'?

> such as `@dhis2/cli-utils-cypress` or `@dhis2/ui-core`

Author: I've changed this, I've added the following comment to the code (`src/utils/codemods/findAvailableCodemodsInNodeModules.js`):

```js
/**
 * Tries to find the following dirs that contain .js files:
 *   - ${NODE_MODULES}/{{folder}}/codemods/
 *   - ${NODE_MODULES}/{{namespace}}/{{folder}}/codemods/
 *
 * @param {Object} paths
 * @returns {String[]}
 */
```

Will update the docs now
